# cs4775-phylo

## Overview
In this repository, you will find all the code and datasets neccessary to reproduce the results presented in "Comparative Evaluation of Phylogenetic Reconstruction Methods for Coronavirus Genomes" (CS 4775 Final Project, Fall 2025). Here is an overview of contents of this repo:

- ```/trees``` directory: we include all
the fasta files with the DNA sequences as well as the generated trees in Newick format. 
-Our Python reimplementation of Fitch's algorithm ``fitchs_algorithm.py`` to calculate parsimony score for a given tree (Newick format)

## Set Up
All the code in this repository is in Python, and uses a couple pip packages which can be installed using our requirements.txt file.

`pip install -r requirements.txt`.

## Data Details
- ``trees/paper-*`` files are Newick formats for trees generated by a paper (i.e. not directly generated by us). 3 files.
- ``trees/{method}-{seq_data}-tree.tre`` are Newick format trees which were generated by ``method`` (BEAST, FastTree, IQ-Tree) on ``seq_data`` (prelim, 252, full) in increasing size. 9 files.

These 12 tree files are the ones for which parsimony scores and RF-distances are reported in Tables 1 and 2 in our final report.

## Running Fitch's Algorithm
To run Fitch's algorithm, use the following terminal command
```
python fitchs_algorithm.py --newick-tree "trees-{newick-file}" --fasta-file "trees/252-fasta-input-file.fas" —-normalize-labels
```

## Calculating RF-Distances
To calculate RF-Distance between `tree1` and `tree2`, run:

``python rf_distance.py --t1 "trees/{tree1} --t2 "trees/{tree2}``.

Note that this calculates unweighted, symmetric RF-distance. To reproduce the results of Table 2 in our report (pairwise distances for all pairs of trees), you can run the Bash script `sh ./calc_rf_dist.sh`.

## Authors
- Deniz Bölöni-Turgut (Computer Science, 2026, db823)
- EJ Hunt (Operations Research and Information Engineering, 2026, ekh56)
- Mac Turner (Computer Science, 2026, mbt52)
- Leane Ying (Computer Science, 2026, ly374)